apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/derived/DerivedDbTable.java

```diff
@@ -154,8 +154,70 @@ public  DatabaseManager getDatabaseManager() {
         }
     }
 
+    @Override
+    public DerivedTableData<T> getAllByDbId(MinMaxDbId minMaxDbId, int limit) throws SQLException {
+        Objects.requireNonNull(minMaxDbId, "minMaxDbId is NULL");
+        TransactionalDataSource dataSource = databaseManager.getDataSource();
+        try (Connection con = dataSource.getConnection();
+             PreparedStatement pstmt = con.prepareStatement("select * from " + table + " where db_id >= ? and db_id < ? limit ?")) {
+            pstmt.setLong(1, minMaxDbId.getMinDbId());
+            pstmt.setLong(2, minMaxDbId.getMaxDbId());
+            pstmt.setLong(3, limit);
+            List<T> values = new ArrayList<>();
+            long dbId = -1;
+            try (ResultSet rs = pstmt.executeQuery()) {
+                while (rs.next()){
+                    values.add(load(con, rs, null));
+                    dbId = rs.getLong("db_id");
+                }
+            }
+            return new DerivedTableData<>(values, dbId);
+        }
+    }
+
     protected abstract T load(Connection con, ResultSet rs, DbKey dbKey) throws SQLException;
 
+    @Override
+    public ResultSet getRangeByDbId(Connection con, PreparedStatement pstmt,
+                                    MinMaxDbId minMaxDbId, int limit) throws SQLException {
+        Objects.requireNonNull(con, "connnection is NULL");
+        Objects.requireNonNull(pstmt, "prepared statement is NULL");
+        Objects.requireNonNull(minMaxDbId, "minMaxDbId is NULL");
+        try {
+            pstmt.setLong(1, minMaxDbId.getMinDbId());
+            pstmt.setLong(2, minMaxDbId.getMaxDbId());
+            pstmt.setLong(3, limit);
+            return pstmt.executeQuery();
+        } catch (SQLException e) {
+            throw e;
+        }
+    }
+
+    @Override
+    public MinMaxDbId getMinMaxDbId(int height) throws SQLException {
+        // select MIN and MAX dbId values in one query
+        String selectMinSql = String.format("SELECT IFNULL(min(DB_ID), 0) as min_DB_ID, " +
+                "IFNULL(max(DB_ID), 0) as max_DB_ID, IFNULL(count(*), 0) as count from %s where HEIGHT <= ?",  table);
+        long dbIdMin = -1;
+        long dbIdMax = -1;
+        MinMaxDbId minMaxDbId = new MinMaxDbId();
+        TransactionalDataSource dataSource = databaseManager.getDataSource();
+        try (Connection con = dataSource.getConnection();
+             PreparedStatement pstmt = con.prepareStatement(selectMinSql)) {
+            pstmt.setInt(1, height);
+            try (ResultSet rs = pstmt.executeQuery()) {
+                if (rs.next()) {
+                    dbIdMin = rs.getLong("min_db_id");
+                    dbIdMax = rs.getLong("max_db_id");
+                    long rowCount = rs.getLong("count");
+                    minMaxDbId = new MinMaxDbId(dbIdMin - 1, dbIdMax + 1); // plus/minus one in Max/Min value
```

Comment:
Why -1?

Reply:
That works good on pagination flow as we retrieve lower bound value in on first loop:
`... where db_id > ? and db_id < ? limit ?`